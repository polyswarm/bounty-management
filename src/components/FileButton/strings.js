const strings = {
  selectFile: 'Select Files to Upload!',
  send: 'Upload',
};
export default strings;
