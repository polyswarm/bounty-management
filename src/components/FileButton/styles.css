.hidden {
    width: 0.1px;
    height: 0.1px;
    overflow: hidden;
    opacity: 0;
    position: absolute;
    z-index: -1;
}

.hidden + label {
  position: relative;
  display: inline-block;
  height: 40px;
  max-width: 350px;
  margin-top: 20px;
  padding: 0 40px;
  background-color: rgb(46, 255, 109);
  color: #8500ff;
  border: 0;
  border-radius: 20px;
  text-shadow: none;
  text-transform: uppercase;
  text-decoration: none;
  line-height: 40px;
  font-size: 16px;
  font-family: 'fabrick', sans-serif;
  letter-spacing: 0.15em;
  transition: background-color 0.25s ease;
  cursor: pointer;
  z-index: 1;
}

.hidden + label:hover,
.hidden + label:active,
.hidden + label.selected {
  background-color: #eeeeee;
  animation-name: hover;
  animation-duration: .5s;
}

@keyframes hover {
  0% {background-color: rgb(46, 255, 109);}
  100% {background-color: #eeeeee}
}
