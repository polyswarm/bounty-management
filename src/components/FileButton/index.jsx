// Vendor imports
import React, {Component} from 'react';
// Component imports
import './styles.css';
import strings from './strings';

class FileButton extends Component {
  constructor(props) {
    super(props);
    this.onFileChanged = this.onFileChanged.bind(this);
  }

  render() {
    return(
      <form>
        <input id='file' className='hidden' type='file' onChange={this.onFileChanged} />
        <label htmlFor='file'>{strings.selectFile}</label>
      </form>
    );
  }

  onFileChanged(event) {
    const {onFileSelected} = this.props;
    if (onFileSelected && event.target.files && event.target.files.length >= 1) {
      onFileSelected(event.target.files[0]);
    }
  }
}
export default FileButton;
