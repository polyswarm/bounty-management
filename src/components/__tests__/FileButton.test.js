import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import FileButton from '../FileButton';

it('renders without crashing', () => {
  const wrapper = render(<FileButton />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('calls the provided fileSelected function', () => {
  // arramnge
  const fileSelected = jest.fn();
  const wrapper = mount(
    <FileButton onFileSelected={fileSelected}/>
  );
  const instance = wrapper.instance();
  const event = {
    target: {
      value: 'fileName'
    }
  };
  // act
  instance.onFileChanged(event);

  // assert
  expect(fileSelected).toHaveBeenCalled();
});
