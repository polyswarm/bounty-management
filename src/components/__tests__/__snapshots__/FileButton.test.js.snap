// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<form>
  <input
    class="hidden"
    id="file"
    type="file"
  >
    
  </input>
  <label
    for="file"
  >
    Select Files to Upload!
  </label>
</form>
`;
